### PR TITLE
Link story fix

### PR DESCRIPTION
**Pull request by takarzyna**
A fix proposition for Link component story, as well as other links in the Storybook - if we don't want them to throw an error after clicking.
Based on: https://github.com/storybooks/react-storybook/issues/485#issuecomment-249401443

Right now I don't think we can use it like react-router's Link and avoid the need of specifying labels in the "label" parameter, though.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -123,6 +122,7 @@
     "webpack-node-externals": "^1.5.4"
   },
   "dependencies": {
+    "@kadira/storybook": "^2.21.0",
```
Comment (takarzyna) [inline package.json]: required by changes in storybook/rr.js and storybook/webpack.config.js

File: src/client/components/NotificationList/NotificationList.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import React from 'react'
 import { List, ListItem } from 'material-ui/List'
 import Divider from 'material-ui/Divider'
-import { Link } from 'react-router'
+import Link from 'components/Link'
```
Comment (takarzyna) [inline src/client/components/NotificationList/NotificationList.js]: the changes in this file were done mostly to test using link in another component. I can drop the changes in this file if you wish.

File: storybook/webpack.config.js [hunk 3]
```diff
@@ -1,26 +1,13 @@
...
+// load the default config generator.
+const genDefaultConfig = require('@kadira/storybook/dist/server/config/defaults/webpack.config.js')
 
-// IMPORTANT
-// When you add this file, we won't add the default configurations which is similar
-// to "React Create App". This only has babel loader to load JavaScript.
+module.exports = function ModifyConfig(config, env) {
+  const newConfig = genDefaultConfig(config, env)
 
-module.exports = {
```
Comment (takarzyna) [inline storybook/webpack.config.js]: I tried adding the loaders to the config, but they produce errors. I don't see where could they be needed. Are they needed?

Comment (kkateq) [inline storybook/webpack.config.js]: what exactly doesn't work?

File: storybook/webpack.config.js [hunk 4]
```diff
@@ -1,26 +1,13 @@
-// you can use this file to add your custom webpack plugins, loaders and anything you like.
-// This is just the basic way to add addional webpack configurations.
-// For more information refer the docs: https://goo.gl/qPbSyX
+// load the default config generator.
+const genDefaultConfig = require('@kadira/storybook/dist/server/config/defaults/webpack.config.js')
```
Comment (kkateq) [inline storybook/webpack.config.js]: We shouldn't add the default configurations here, you should be able resolve aliases without default storybook config

File: storybook/webpack.config.js [hunk 5]
```diff
@@ -1,26 +1,13 @@
...
-        test: /\.png$/,
-        loader: 'url-loader',
-        query: { mimetype: 'image/png' },
-      },
-    ],
-  },
+  // this is used by the custom `rr.js` module
+  newConfig.resolve.alias['react-router-original'] = require.resolve('react-router')
+  // this `rr.js` will replace the Link with a our own mock component
+  newConfig.resolve.alias['react-router'] = require.resolve('./rr.js')
```
Comment (kkateq) [inline storybook/webpack.config.js]: Could we rename `rr.js` file to something  maybe `react-router.js` ?

File: storybook/webpack.config.js (hunk 3 above)
Comment (takarzyna) [inline storybook/webpack.config.js]: running the storybook. A lot of `Module build failed: Unknown word` errors appear.

This is the code I use to add the loaders:
```
newConfig.module.loaders.push({
    test: /\.css$/,
    loader: 'style!css?localIdentName=[local]___[hash:base64:5]',
  })
  newConfig.module.loaders.push({
    test: /\.png$/,
    loader: 'url-loader',
    query: { mimetype: 'image/png' },
  })
```

File: storybook/webpack.config.js (hunk 4 above)
Comment (takarzyna) [inline storybook/webpack.config.js]: Oh, I just discovered that the errors I was getting earlier disappear if I return the config without creating a `const newConfig`. I'll prepare a new commit

**Pull request merged**